onboard/catkin_ws/src/offboard_comms/Arduino Sketchbooks/ThrusterServoArduino/MultiplexedServo.cpp

```diff
@@ -2,15 +2,51 @@
 #include <Wire.h>
 #include "Adafruit_PWMServoDriver.h"
 #include "MultiplexedServo.h"
-#include "MultiplexedObject.h"
 
-MultiplexedServo::MultiplexedServo(Adafruit_PWMServoDriver *_multiplexer,int _num):MultiplexedObject(_multiplexer, _num){}
+MultiplexedServo::MultiplexedServo(Adafruit_PWMServoDriver *_multiplexer):multiplexer(_multiplexer){
+  is_attached = false;
+}
+
+MultiplexedServo::~MultiplexedServo(){
+  detach();
+}
+
+void MultiplexedServo::attach(uint8_t _pin){
+  attach(_pin, 544, 2400);
+}
+
+void MultiplexedServo::attach(uint8_t _pin, uint16_t _out_min, uint16_t _out_max){
+  attach(_pin, _out_min, _out_max, 0, 180);
+}
+
+void MultiplexedServo::attach(uint8_t _pin, uint16_t _out_min, uint16_t _out_max, int16_t _in_min, int16_t _in_max){
```

Comment:
This is somewhat counterintuitive as an API for a servo class, and makes the program harder to read because both the thrusters and servos are now represented as `MultiplexedServo` objects.

The bigger issue though is that you are now storing these range values inside every object, which makes them much larger than they need to be. This would make sense if there was a need to dynamically set the input and output ranges, however, in this case, we have that information at compile time.

It's also important that it be easy to swap out the `MultiplexedServo` class for the built in `Servo` class because that will make it a lot easier to run simple tests. Keeping almost the same API is a big part of that.

I'd recommend keeping the `MultiplexedBasicEsc` class and just having it wrap around a `MultiplexedServo` object, and the former can store the specific constants needed to do conversions.